### PR TITLE
Add option to set delimeter for environment variables

### LINKED ISSUE #14: Add option to set delimeter for environment variables
Hi, me again, I have this case in which I'm attempting to set an env variable in the CF which contains comma (",") characters.
This is what comes up in the log:

```
2020/07/15 20:11:49 Drone-GCF Plugin  version: [not-tagged]   hash:    date: 2020-07-10-16:51:55
2020/07/15 20:11:49 Using project ID: xxxxxxxxxxx
Google Cloud SDK 290.0.1
alpha 2020.04.24
beta 2020.04.24
bq 2.0.56
core 2020.04.24
gsutil 4.49
kubectl 2020.04.24
Activated service account credentials for: [xxxx@xxxxx.iam.gserviceaccount.com]
ERROR: (gcloud.functions.deploy) argument --set-env-vars: Bad syntax for dict arg: [XNuWskz]. Please see `gcloud topic flags-file` or `gcloud topic escaping` for information on providing list or dictionary flag values with special characters.
Usage: gcloud functions deploy (NAME : --region=REGION) [optional flags]
  optional flags may be  --allow-unauthenticated | --clear-env-vars |
                         --clear-labels | --clear-max-instances |
                         --clear-vpc-connector | --egress-settings |
                         --entry-point | --env-vars-file | --help |
                         --ignore-file | --ingress-settings | --max-instances |
                         --memory | --region | --remove-env-vars |
                         --remove-labels | --retry | --runtime |
                         --service-account | --set-env-vars | --source |
                         --stage-bucket | --timeout | --trigger-bucket |
                         --trigger-event | --trigger-http | --trigger-resource |
                         --trigger-topic | --update-env-vars | --update-labels |
                         --vpc-connector

For detailed information on this command and its flags, run:
  gcloud functions deploy --help
2020/07/15 20:11:50 runConfig() err: error: exit status 2
```

if you go and check those  `gcloud topic flags-file` or `gcloud topic escaping`, you'll find two posibilities to allow the use of commas and other special characters in the values for a "dictionary flag", which happens to be the case for `---set-env-vars`.
To sum up, what needs to happen is that when the command is building it's flags, it should allow to build it like this:
`--set-env-vars=^DELIM^`, maybe pass that DELIM in a `environment_delimiter` setting?

That would allow us to have commas (and any other character) in the values for our environment variables. like so:
```yaml
- name: deploy-cloud-functions
    image: oliver006/drone-gcf
    settings:
      action: deploy
      project: xxxxxx
      token:
        from_secret: token
      functions:
        - MyFunction:
          - trigger: http
            runtime: nodejs10
            memory: 2048MB
            environment_delimiter: ";"
            environment:
              - SOME_ENV: "some,value,with,commas"
                SOME_OTHER_ENV: "someValueWithoutCommas"
```
which should result in this flag:
`--set-env-vars=^;^SOME_ENV=some,value,with,commas;SOME_OTHER_ENV=someValueWithoutCommas`

### PR DESCRIPTION
**Pull request by leanazulyoro**
closes #14 

This is my first golang code ever, so please check it carefully.
The intention is to have the default delimiter to be the one that was defined already (":|:"), and if the environment_delimiter settings is defined, then use that one instead.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -300,7 +301,11 @@ func CreateExecutionPlan(cfg *Config) (Plan, error) {
 					}
 				}
 
-				envStr := "^:|:^" + strings.Join(e, ":|:")
+				envsDelimiter := ":|:"
```
Comment (oliver006) [inline main.go:304]: Can you make this a const called `defaultEnvVarDelimiter` or similar and move up to the block near line 55?

File: main.go [hunk 2]
```diff
@@ -300,7 +301,11 @@ func CreateExecutionPlan(cfg *Config) (Plan, error) {
...
+				if f.EnvironmentDelimiter != "" {
```
Comment (oliver006) [inline main.go:305]: How about checking and setting the delimiter up here: https://github.com/oliver006/drone-gcf/blob/58f0568a73bdd522a0d5d10cac36b51c501af61a/main.go#L137

and then just use it here?

**Comment by leanazulyoro** [comment]
@oliver006 I've adjusted according to your comments, but the pipeline is not running correctly. I've ran tests locally and they pass. Do you think this is a pipeline issue or it has something to do with the changes I've introduced?

**Comment by oliver006** [comment]
Looks like the go fmt check fails - can you make sure all the files are formatted correctly?

**Comment by leanazulyoro** [comment]
@oliver006 great thanks!, I formatted it with `go fmt`. Let me know if there's anything missing.

**Comment by oliver006** [comment]
Released as https://github.com/oliver006/drone-gcf/releases/tag/v1.9.0

**Pull request merged**